src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,72 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+export enum ForFilteredTodos {
+  all = 'all',
+  noCompleted = 'noCompleted',
+  completed = 'completed',
+}
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string;
+  getSetQuery: (value: string) => void;
+  selectedFilter: string;
+  getSelectedFilter: (value: ForFilteredTodos) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  query,
+  getSetQuery,
+  selectedFilter,
+  getSelectedFilter,
+}) => {
+  const reset = () => {
+    getSetQuery('');
+    getSelectedFilter(ForFilteredTodos.all);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={selectedFilter}
+            onChange={(event) => {
+              getSelectedFilter(event.target?.value as ForFilteredTodos);
+            }}
+          >
+            {Object.entries(ForFilteredTodos).map(([key, value]) => (
+              <option value={value} key={key}>
+                {value}
+              </option>
+            ))}
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={(event) => getSetQuery(event.target.value)}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          {query !== '' && (
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              aria-label="Clear Search"
+              onClick={reset}
+            />
```

Comment:
https://github.com/mate-academy/react_dynamic-list-of-todos/pull/1868#discussion_r1329859219 - good point.